cts.json

```diff
@@ -3819,6 +3819,48 @@
       "selector": "$[-1:-10:-231584178474632390847141970017375815706539969331281128078915168015826259279872]",
       "invalid_selector": true
     },
+    {
+      "name": "functions, arguments, parenthesized expression",
+      "selector": "$[?match((@.a), ('a.*'))]",
+      "document": [
+        {
+          "a": "ab"
+        },
+        {
+          "a": "ba"
+        }
+      ],
+      "result": [
+        {
+          "a": "ab"
+        }
+      ]
+    },
+    {
+      "name": "functions, arguments, function expression",
+      "selector": "$.values[?match(@.a, value($..['regex'])]",
+      "document": {
```

Comment:
For an invalid selector, the document is irrelevant and should be omitted.

Reply:
Unless I've misunderstood (quite possible :smiley:), `$.values[?match(@.a, value($..['regex'])]` looks like a valid query, and the given result is correct when applied to that document.

1. `$..['regex']` evaluates to a nodelist, of `NodesType`, which happens to contain a single node.
2. `value()` accepts a `NodesType` argument and returns a `ValueType`. In this case `value($..['regex'])` evaluates to `"a.*"`, because its argument contains a single node.
3. `match()` is called with the `ValueType` implicitly converted from the singular query `@.a`, and the `ValueType`  (`"a.*"`) from the call to `value()`.

Reply:
The idea is good, but the path is invalid.  It needs another `)`

```jsonpath
$.values[?match(@.a, value($..['regex']))]
```

Tested at https://json-everything.net/json-path.

Reply:
Thanks. Looks like I have some issues with unbalanced bracket detection.